File: src/main/java/com/leijendary/spring/microservicetemplate/document/SampleDocument.java
package com.leijendary.spring.microservicetemplate.document;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.data.elasticsearch.annotations.Document;
import org.springframework.data.elasticsearch.annotations.Field;
import org.springframework.data.elasticsearch.annotations.FieldType;

import javax.persistence.Id;
import java.time.OffsetDateTime;

import static org.springframework.data.elasticsearch.annotations.DateFormat.date_time;
import static org.springframework.data.elasticsearch.annotations.FieldType.Text;

@Data
@EqualsAndHashCode(callSuper = true)
@Document(indexName = "sample")
public class SampleDocument extends LocalizedDocument<SampleTranslationDocument> {

    @Id
    @Field(type = FieldType.Long)
    private long id;

    @Field(type = Text)
    private String column1;

    @Field(type = FieldType.Integer)
    private int column2;

    @Field(type = FieldType.Date, format = date_time)
    private OffsetDateTime createdDate;
}

File: src/main/kotlin/com/leijendary/spring/template/document/SampleDocument.kt
package com.leijendary.spring.template.document

import com.leijendary.spring.template.core.document.LocalizedDocument
import org.springframework.data.annotation.Id
import org.springframework.data.elasticsearch.annotations.DateFormat.date_hour_minute_second_millis
import org.springframework.data.elasticsearch.annotations.Document
import org.springframework.data.elasticsearch.annotations.Field
import org.springframework.data.elasticsearch.annotations.FieldType
import org.springframework.data.elasticsearch.annotations.FieldType.*
import org.springframework.data.elasticsearch.annotations.FieldType.Date
import org.springframework.data.elasticsearch.annotations.FieldType.Double
import org.springframework.data.elasticsearch.annotations.Setting
import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.*

@Document(indexName = "sample")
@Setting(settingPath = "/elasticsearch/ngram-analyzer.settings.json")
data class SampleDocument(
    @Id
    @Field(type = Keyword)
    var id: UUID,

    @Field(type = Text, analyzer = "ngram_analyzer", searchAnalyzer = "standard")
    var column1: String,

    @Field(type = FieldType.Long)
    var column2: Long,

    @Field(type = Double)
    var amount: BigDecimal,

    @Field(type = Date, format = [date_hour_minute_second_millis])
    var createdAt: LocalDateTime
) : LocalizedDocument<SampleTranslationDocument>()